### PR TITLE
Groups datasource (plural)

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/99

## Changes
- added datasource for groups (plural)
- moved client errors in separate file

### CODE REVIEW

File: twingate/datasource_groups.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the Group",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the Group",
```
Comment (vmanilo) [inline twingate/datasource_groups.go:63]: @alexmensch please check descriptions

File: twingate/datasource_groups.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func datasourceGroups() *schema.Resource {
+	return &schema.Resource{
+		Description: "Groups are how users are authorized to access Resources. For more information, see Twingate's [documentation](https://docs.twingate.com/docs/groups).",
+		ReadContext: datasourceGroupsRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
```
[the remark this answers is not in the document]
Comment (vmanilo) [reply twingate/datasource_groups.go:46]: ok, will update PR

File: docs/data-sources/groups.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+Groups are how users are authorized to access Resources. For more information, see Twingate's [documentation](https://docs.twingate.com/docs/groups).
+
+## Example Usage
+
+```terraform
+data "twingate_groups" "foo" {
+  name = "<your group's name>"
+}
+```
+
```
Comment (alexmensch) [inline docs/data-sources/groups.md:20]: Let's add a clarification comment below this to give some context as to why this is a plural data source.

`Group names are not constrained to be unique within Twingate, so it is possible that this data source will return multiple list items.`

**Review by alexmensch — CHANGES_REQUESTED** [review]

File: docs/data-sources/groups.md (hunk 3 above)
Comment (vmanilo) [reply docs/data-sources/groups.md:23]: updated docs

Comment (alexmensch) [reply docs/data-sources/groups.md:23]: Ah, I see that this is within the Terraform example itself, and now I understand the `#` comment notation. Originally I was thinking that we should change this to move it up into the general description, but I think this works. Will approve this and the other similar changes and then we can merge everything.

**Review by alexmensch — APPROVED** [review]

**Pull request merged**